docs/guides/authjs-migration.mdx

```diff
@@ -0,0 +1,390 @@
+
+```tsx filename="middleware.ts"
+import { authMiddleware } from "@clerk/nextjs";
+ 
+// This example protects all routes including api/trpc routes
+// Please edit this to allow other routes to be public as needed.
+// See https://clerk.com/docs/references/nextjs/auth-middleware for more information about configuring your Middleware
+export default authMiddleware({});
+ 
+export const config = {
+  matcher: ["/((?!.+\\.[\\w]+$|_next).*)", "/", "/(api|trpc)(.*)"],
+};
+```
+
+#### Control access to your app with Clerk's components
+
+To conditionally render UI when the user is signed in, wrap it with [`<SignedIn>`](/docs/components/control/signed-in).
+
+To conditionally render UI when the user is _not_ signed in, wrap it with [`<SignedOut>`](/docs/components/control/signed-out).
+
+```tsx filename="app/page.tsx"
+import { SignedIn, SignedOut } from "@clerk/nextjs";
+ 
+export default function Home() {
+  return (
+    <div>
+      <SignedOut>
+        <p>This content is public. Only signed out users can see this.</p>
+      </SignedOut>
+      <SignedIn>
+        <p>This content is private. Only signed in users can see this.</p>
+      </SignedIn>
+    </div>
+  )
+}
+```
+
+### Read user and session data
+
+#### Server side
+
+Replace any Auth.js `getServerSession(req, res, authOptions)` with Clerk's helpers.
+
+Let's see what your code might look like with Auth.js's `getServerSession()`:
+
+```tsx filename="app/page.tsx"
+import { getServerSession } from "next-auth";
+import { authOptions } from "@/pages/api/auth/[...nextauth]";
+
+export default function Page() {
+  const data = await getServerSession(authOptions);
+  console.log(data)
+
+  return (
+    <p>Home Page</p>
+  )
+}
+```
+
+Now let's see what your code should look like when replacing `getServerSession` with Clerk's helpers:
+
+<Tabs type="installer" items={["App Router", "Pages Router"]}>
+  <Tab>
+  You can replace Auth.js's `setServerSession` with Clerk's [`auth()`](/docs/references/nextjs/auth#auth) helper in order to read your user data.
+
+  ```tsx failename="app/page.tsx"
+  import { auth, currentUser } from "@clerk/nextjs";
+  
+  export default async function Page() {
+    const { userId } = auth();
+    console.log(userId)
+    
+    return (
+      <p>Home Page</p>
+    )
+  }
+  ```
+  </Tab>
+
+  <Tab>
+  You can replace Auth.js's `setServerSession` with Clerk's [`getAuth()`](/docs/references/nextjs/get-auth#get-auth) helper in order to read your user data.
+
+  ```tsx filename="pages/index.tsx"
+  export async function getServerSideProps(context) {
+    const session = getAuth(context.req);
+    
+    return { props: { ...buildClerkProps(ctx.req) } };
+  };
+  ```
+  </Tab>
+</Tabs>
+
+#### Client Side
+
+Replace Auth.js's `useSession()` hook with Clerk's hooks.
+
+Let's see what your code might look like with Auth.js's `useSession()`:
```

Comment:
We have App Router & Pages Router examples all thru out this guide. We should stay consistent and show here what the `useSession()` usage would like in their Pages Router app

Reply:
Its the basically the same. Remove `'use client'` and change the filename. I didn't include as we are working to de-dupe code.

Reply:
ohh you know what you're right